include/lcp-catlistdisplayer.php

```diff
@@ -317,79 +311,42 @@ private function content_getter($type, $post, $tag = null, $css_class = null) {
+      // Default wrapper behavior not supported here,
+      // class is only used inside the <a> element.
+      $css_class = null;
     }
-    return $this->assign_style($info, $tag, $css_class);
-  }
-
-  private function get_conditional_title(){
-    if(!empty($this->params['conditional_title_tag']))
-      $tag = $this->params['conditional_title_tag'];
-    else
-      $tag = 'h3';
-    if(!empty($this->params['conditional_title_class']))
-      $class = $this->params['conditional_title_class'];
-    else
-      $class = '';
-
-    return $this->assign_style($this->catlist->get_conditional_title(), $tag, $class);
-  }
-
-  private function get_custom_fields($single){
-    if(!empty($this->params['customfield_display'])){
-      $info = $this->catlist->get_custom_fields($this->params['customfield_display'], $single->ID);
-      if(empty($this->params['customfield_tag']) || $this->params['customfield_tag'] == null){
-        $tag = 'div';
-      } else {
-        $tag = $this->params['customfield_tag'];
-      }
-
-      if(empty($this->params['customfield_class']) || $this->params['customfield_class'] == null){
-        $css_class = 'lcp_customfield';
-      } else {
-        $css_class = $this->params['customfield_class'];
-      }
-
-      $final_info = '';
-      if(!is_array($info)){
-        $final_info = $this->assign_style($info, $tag, $css_class);
-      }else{
-        if($this->params['customfield_display_separately'] != 'no'){
-          foreach($info as $i)
-            $final_info .= $this->assign_style($i, $tag, $css_class);
-        }else{
-          $one_info = implode($this->params['customfield_display_glue'], $info);
-          $final_info = $this->assign_style($one_info, $tag, $css_class);
-        }
-      }
-      return $final_info;
-    }
-  }
-
-  private function get_date($single, $tag = null, $css_class = null){
-    $info = $this->catlist->get_date_to_show($single);
-
-    if ( !empty($this->params['link_dates']) && ( 'yes' === $this->params['link_dates'] || 'true' === $this->params['link_dates'] ) ):
-      $info = $this->get_post_link($single, $info);
-    endif;
-
-    $info = ' ' . $info;
-    return $this->assign_style($info, $tag, $css_class);
-  }
-
-  private function get_modified_date($single, $tag = null, $css_class = null){
-    $info = " " . $this->catlist->get_modified_date_to_show($single);
-    return $this->assign_style($info, $tag, $css_class);
-  }
-
-  private function get_thumbnail($single, $tag = null){
-    if ( !empty($this->params['thumbnail_class']) ) :
-      $lcp_thumb_class = $this->params['thumbnail_class'];
-      $info = $this->catlist->get_thumbnail($single, $lcp_thumb_class);
-    else:
-      $info = $this->catlist->get_thumbnail($single);
-    endif;
-
-    return $this->assign_style($info, $tag);
+    return $this->wrapper->wrap($info, $tag, $css_class);
   }
 
   private function get_post_link($single, $text, $class = null){
```

Comment:
I left `get_post_link`, `get_post_title` and `lcp_title_limit` as they are for now. Mostly because they work differently than the stuff above.